### PR TITLE
[stable/fluent-bit] Stackdriver backend

### PR DESCRIPTION
**Pull request by chrissng**
#### What this PR does / why we need it:

This PR implements the Stackdriver backend to allow Fluent-bit to log outputs to the Google Cloud Stackdriver. Chart implementation follow the official documentation in https://docs.fluentbit.io/manual/v/1.2/output/stackdriver

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

**Comment by chrissng** [comment]
Hello @edsiper and @Towmeykaw, would you be so kind as to review this PR? Thank you

**Comment by chrissng** [comment]
Hello @kfox1111 and @hectorj2f , would you be so kind as to review this PR? Thank you

File: stable/fluent-bit/templates/daemonset.yaml [hunk 1]
```diff
@@ -97,9 +97,14 @@ spec:
...
 {{- end }}
+{{- if and .Values.backend.type "stackdriver" .Values.backend.stackdriver.google_service_credentials }}
+        - name: fluentbit-secret
+          mountPath: /secure/google_service_credentials.json
+          subPath: google_service_credentials.json
+{{- end }}
 {{- end }}
 {{- if .Values.backend.es.tls_ca }}
-        - name: es-tls-secret
+        - name: fluentbit-secret
```
Comment (hectorj2f) [inline stable/fluent-bit/templates/daemonset.yaml]: why do you change the name of this secret ? let's keep the names as before

File: stable/fluent-bit/templates/daemonset.yaml [hunk 2]
```diff
@@ -141,11 +146,9 @@ spec:
           path: /etc/machine-id
           type: File
     {{- end }}
-{{- if .Values.backend.es.tls_ca }}
-      - name: es-tls-secret
+      - name: fluentbit-secret
```
Comment (hectorj2f) [inline stable/fluent-bit/templates/daemonset.yaml]: also, this doesn't sound correct

File: stable/fluent-bit/templates/daemonset.yaml [hunk 3]
```diff
@@ -97,9 +97,14 @@ spec:
           mountPath: /fluent-bit/etc/parsers_custom.conf
           subPath: parsers.conf
 {{- end }}
+{{- if and .Values.backend.type "stackdriver" .Values.backend.stackdriver.google_service_credentials }}
+        - name: fluentbit-secret
```
Comment (hectorj2f) [inline stable/fluent-bit/templates/daemonset.yaml]: where is its respective volume that mounts this file from the host machine ?

File: stable/fluent-bit/values.yaml [hunk 4]
```diff
@@ -80,6 +80,10 @@ backend:
     tls_verify: "off"
     tls_debug: 1
     message_key: "kubernetes"
+  stackdriver:
```
Comment (hectorj2f) [inline stable/fluent-bit/values.yaml]: `stackdriver: {}` should be the right value

**Review by hectorj2f — CHANGES_REQUESTED** [review]

File: stable/fluent-bit/templates/daemonset.yaml (hunk 1 above)
Comment (chrissng) [reply stable/fluent-bit/templates/daemonset.yaml]: I don't see the need to create another Secrets volume just to use the google cloud credentials.

Hence I renamed the existing Secrets volume name to be reference for both the google cloud credentials and the SSL cert used by elasticsearch.

File: stable/fluent-bit/templates/daemonset.yaml (hunk 3 above)
Comment (chrissng) [reply stable/fluent-bit/templates/daemonset.yaml]: It is a secret mounted as a volume https://github.com/helm/charts/pull/15549/files#diff-f455fb5ea559681c3a3dd88acfd3e6aaR149

File: stable/fluent-bit/templates/daemonset.yaml (hunk 2 above)
Comment (chrissng) [reply stable/fluent-bit/templates/daemonset.yaml]: Any suggestions?

Comment (chrissng) [reply stable/fluent-bit/templates/daemonset.yaml]: I've separated secrets into two - one used by es (also see pr https://github.com/helm/charts/pull/14849/files which added more overrides for es secrets) and another for gcp.

File: stable/fluent-bit/templates/daemonset.yaml (hunk 1 above)
Comment (chrissng) [reply stable/fluent-bit/templates/daemonset.yaml]: Changed back. See comment https://github.com/helm/charts/pull/15549#discussion_r318852224

**Comment by zanhsieh** [comment]
@chrissng 
Could you resolve the conflict of these two files?
stable/fluent-bit/Chart.yaml
stable/fluent-bit/templates/daemonset.yaml

**Comment by chrissng** [comment]
Resolved.

**Comment by zanhsieh** [comment]
@kfox1111
@edsiper
@hectorj2f
@Towmeykaw
Would you mind to LGTM this PR please? Thank you guys.

File: stable/fluent-bit/templates/daemonset.yaml [hunk 5]
```diff
@@ -163,6 +168,11 @@ spec:
         secret:
           secretName: {{ .Values.backend.es.tls_secret | default ( print ( include "fluent-bit.fullname" . ) "-es-tls-secret" ) | quote }}
 {{- end }}
+{{- if and .Values.backend.type "stackdriver" .Values.backend.stackdriver.google_service_credentials }}
+      - name: gcp-secret
```
Comment (hectorj2f) [inline stable/fluent-bit/templates/daemonset.yaml]: Can we name this secret as gcp-stackdriver-secret?

File: stable/fluent-bit/templates/daemonset.yaml [hunk 6]
```diff
@@ -114,6 +114,11 @@ spec:
           mountPath: /fluent-bit/etc/parsers_custom.conf
           subPath: parsers.conf
 {{- end }}
+{{- if and .Values.backend.type "stackdriver" .Values.backend.stackdriver.google_service_credentials }}
+        - name: gcp-secret
```
Comment (hectorj2f) [inline stable/fluent-bit/templates/daemonset.yaml]: Same as below

File: stable/fluent-bit/templates/secret.yaml [hunk 7]
```diff
@@ -12,3 +13,18 @@ type: Opaque
 data:
   es-tls-ca.crt: {{ .Values.backend.es.tls_ca | b64enc | quote }}
 {{- end }}
+{{- if .Values.backend.stackdriver.google_service_credentials }}
+---
```
Comment (hectorj2f) [inline stable/fluent-bit/templates/secret.yaml:17]: Shouldn't you check if the backend type is stackdriver?

Comment (chrissng) [reply stable/fluent-bit/templates/secret.yaml:17]: Got it. Fixed.

File: stable/fluent-bit/templates/daemonset.yaml (hunk 5 above)
Comment (chrissng) [reply stable/fluent-bit/templates/daemonset.yaml]: Done

File: stable/fluent-bit/templates/daemonset.yaml (hunk 6 above)
Comment (chrissng) [reply stable/fluent-bit/templates/daemonset.yaml]: Done

**Review by hectorj2f — COMMENTED** [review]
/lgtm

**Pull request merged**